codegenerator/cli/templates/static/codegen/src/EventProcessing.res

```diff
@@ -254,24 +254,27 @@ let addEventToRawEvents = (
 }
 
 let runEventHandler = (
-  eventBatchQueueItem: Types.eventBatchQueueItem,
-  ~loaderHandler: Types.HandlerTypes.loaderHandler<_>,
+  eventItem: Types.eventItem,
+  ~loader,
+  ~handler,
   ~inMemoryStore,
   ~logger,
   ~loadLayer,
   ~shouldSaveHistory,
 ) => {
   open ErrorHandling.ResultPropogateEnv
   runAsyncEnv(async () => {
-    let contextEnv = ContextEnv.make(~eventBatchQueueItem, ~logger)
-    let {loader, handler} = loaderHandler
+    let contextEnv = ContextEnv.make(~eventItem, ~logger)
 
     //Include the load in time before handler
     let timeBeforeHandler = Hrtime.makeTimer()
 
-    let loaderReturn =
-      (await runEventLoader(~contextEnv, ~loader, ~inMemoryStore, ~loadLayer))->propogate
-
+    let loaderReturn = switch loader {
+      | Some(loader) =>
+        (await runEventLoader(~contextEnv, ~loader, ~inMemoryStore, ~loadLayer))->propogate
+      | None => %raw(`undefined`)
+    }
```

Comment:
Don't run it for handlers without loaders 🚀

Reply:
Nice, interested how this affects performance for handlers without loaders.

Reply:
Maybe ~1ms. I don't think that there is a big difference.

Reply:
Well that could make a big difference. For eg.

This handler runtime totalled over 4 hours during historical sync
![Screenshot 2024-12-05 at 14 18 16](https://github.com/user-attachments/assets/5c269aff-d746-47b8-9b7e-b6d9a8051ab9)

But the average time for that handler was tiny

![Screenshot 2024-12-05 at 14 19 44](https://github.com/user-attachments/assets/02fe9aee-3fb3-4ee3-a8c5-a9f8b9ebbafb)